src/openforms/registrations/contrib/objects_api/models.py

```diff
@@ -28,31 +26,77 @@ class ObjectsAPIConfig(SingletonModel):
     global configuration and defaults
     """
 
-    objects_service = models.OneToOneField(
+    productaanvraag_type = models.CharField(
+        _("Productaanvraag type"),
+        help_text=_(
+            "Description of the 'ProductAanvraag' type. This value is saved in the 'type' attribute of the 'ProductAanvraag'."
+        ),
+        blank=True,
+        max_length=255,
+    )
+    content_json = models.TextField(
+        _("JSON content template"),
+        validators=[
+            DjangoTemplateValidator(
+                backend="openforms.template.openforms_backend",
+            ),
+        ],
+        blank=False,
+        default=get_content_text,
+        help_text=_(
+            "This template is evaluated with the submission data and the resulting JSON is sent to the objects API."
+        ),
+    )
+    payment_status_update_json = models.TextField(
+        _("payment status update JSON template"),
+        validators=[
+            DjangoTemplateValidator(
+                backend="openforms.template.openforms_backend",
+            ),
+        ],
+        blank=False,
+        default=get_payment_status_update_text,
+        help_text=_(
+            "This template is evaluated with the submission data and the resulting JSON is sent to the objects API "
+            "with a PATCH to update the payment field."
+        ),
+    )
+
+    class Meta:
+        verbose_name = _("Objects API configuration")
+
+
+class ObjectsAPIGroupConfig(models.Model):
```

Comment:
can you add a TODO to remove the `null=True` from the fields in Open Forms 3.0? And add it to the ticket for 3.0 deprecations to remove. We have to accept them for now because data is incomplete, but going forward our data model should prevent incomplete configurations :)